pilot/cmd/pilot-agent/main.go

```diff
@@ -85,6 +85,7 @@ var (
 	serviceAccountVar    = env.RegisterStringVar("SERVICE_ACCOUNT", "", "Name of service account")
 	clusterIDVar         = env.RegisterStringVar("ISTIO_META_CLUSTER_ID", "", "")
 	callCredentials      = env.RegisterBoolVar("CALL_CREDENTIALS", false, "Use JWT directly instead of MTLS")
+	enableTokenExchange  = env.RegisterBoolVar("ENABLE_TOKEN_EXCHANGE", false, "Enable token exchange through STS")
```

Comment:
I don't think this is the correct API.

* The option is ignored for google CA randomly
* There is no distinction between CA and XDS, meaning your change will only support google ca+google XDS server. We need to be generic.
* I don't think we should be bolting on yet another environment flag to go with the other 8. The config is already extremely convoluted.

I suggest we iterate on https://docs.google.com/document/d/1bi_wkznNsejh-BgnTta6g7StUOOEzLqDlwIiZ-6jxGI/ for a proper API.

Reply:
- According to existing implementation, for CA flows, the token exchange is always allowed. This PR does not change the existing behavior. Can you advise how the value for CA flows should be configured?
- Do you suggest CA and XDS to use separate flags for enabling token exchange?
- @myidpt @shankgan @howardjohn, can you advise which API to use for this flag?

Reply:
> According to existing implementation, for CA flows, the token exchange is always allowed. This PR does not change the existing behavior. Can you advise how the value for CA flows should be configured?

That is not accurate, there is no configured TokenExchanger when not using GoogleCA

> Do you suggest CA and XDS to use separate flags for enabling token exchange?

Yes, I think there should be a common API for XDS and CA client configuration. When only one is configured, the CA configuration would default to the XDS configuration (to handle the common case in OSS where XDS==CA and all authentication and communicate is the same. We can even share a single tcp connection).
Otherwise, we are just slapping on more an more hacks. for example, does "ENABLE_TOKEN_EXCHANGE" apply to XDS, CA, or both? What about CREDENTIAL_FETCHER_TYPE, CREDENTIAL_IDENTITY_PROVIDER, and PROV_CERT?

Without a consistent API, these are super hard to reason about - for users and developers. And, as seen by this PR, the current status quo does not provide enough flexibility.

Reply:
I agree that having a common API (XDS/SDS) will make readability and usability easier. However, figuring out the common API and implementing them may not happen very quickly? In the interim, perhaps this flag could be differently labelled - perhaps as XDS_ENABLE_TOKEN_EXCHANGE? Correct me if this is incorrect - Even once we unify the API's, we will continue to need two sets of configuration for XDS and SDS, since they may not be hosted on the same service. There are scenarios where the SDS service may require token-exchange, while the XDS service does not.

Reply:
Yes, they require two sets of config. Ideally, we have the exact same schema for both, since they need all the same options potentially. If a user provides only one, we default to using the same config for them *and use the same connection*. This allows real performance improvements by reducing the total number of TCP connections in the cluster.

Reply:
Based on the comments, the PR will be changed to use two flags for XDS and SDS, which unblocks the work of enabling token exchange for XDS in the interim period before the API in  https://docs.google.com/document/d/1bi_wkznNsejh-BgnTta6g7StUOOEzLqDlwIiZ-6jxGI/  is ready. After the API is ready, the two flags can be migrated to conform to the API.

Reply:
@howardjohn @shankgan The PR has been revised based on the comments. Please take another look. Thanks!